rasa/nlu/featurizers/dense_featurizer/lm_featurizer.py

```diff
@@ -42,58 +45,102 @@
 logger = logging.getLogger(__name__)
 
 
-class LanguageModelFeaturizer(DenseFeaturizer):
-    """Featurizer using transformer-based language models.
+class LanguageModelFeaturizerGraphComponent(DenseFeaturizer2, GraphComponent):
+    """A featurizer that uses transformer-based language models.
 
-    The transformers(https://github.com/huggingface/transformers) library
-    is used to load pre-trained language models like BERT, GPT-2, etc.
-    The component also tokenizes and featurizes dense featurizable attributes of
+    This component loads a pre-trained language model
+    from the Transformers library (https://github.com/huggingface/transformers)
+    including BERT, GPT, GPT-2, xlnet, distilbert, and roberta.
+    It also tokenizes and featurizes the featurizable dense attributes of
     each message.
     """
 
-    defaults = {
-        # name of the language model to load.
-        "model_name": "bert",
-        # Pre-Trained weights to be loaded(string)
-        "model_weights": None,
-        # an optional path to a specific directory to download
-        # and cache the pre-trained model weights.
-        "cache_dir": None,
-    }
-
-    @classmethod
-    def required_components(cls) -> List[Type[Component]]:
-        """Packages needed to be installed."""
-        return [Tokenizer]
-
     def __init__(
         self,
-        component_config: Optional[Dict[Text, Any]] = None,
-        skip_model_load: bool = False,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
```

Comment:
```suggestion
```

Reply:
The execution context is used in the superclass constructor call. Should I keep it and remove the others?

Reply:
yep, you can remove the unused ones from the `init` (they only need to stay in the `create` because we need that function to implement the interface defined by the GraphComponent)